File: 0318/SeoyeonHong/1107 리모컨.py
# 채널 N으로 이동하기 위해서 버튼을 최소 몇 번 눌러야하는지 
N = int(input())
M = int(input())
broken = []
if M > 0:
    broken = list(input().split())
min_count = abs(N - 100)

def can_make_number(num):
    for n in num:
        if n in broken:
            return False
    return True

for i in range(abs(N-100)):
    possible = []
    if N-i >= 0 and can_make_number(str(N-i)):
        possible.append(str(N-i))
    if can_make_number(str(N+i)):
        possible.append(str(N+i))
    
    if possible:
        min_count = min(min_count, i + len(possible[0]))

print(min_count)

File: 0318/SoohyunKim/1107_리모컨.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

function solution(input) {
  const N = +input[0]; 
  const M = +input[1]; 
  if (N === 100) return 0;
  
  const broken = M > 0 ? new Set(input[2].split(' ').map(Number)) : new Set();
  
  // +, - 만 사용했을 때
  let answer = Math.abs(N - 100);

  // 숫자 버튼 이용할 때
  for (let i = 0; i < 1_000_000; i++) {
    const str = i.toString();
    if ([...str].every(ch => !broken.has(Number(ch)))) { 
      const pressCount = str.length + Math.abs(i - N);
      answer = Math.min(answer, pressCount);
    }
  }

  return answer;
}

console.log(solution(input));